File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/article/ArticleContract.java
package pl.orbitemobile.wspolnoty.activities.article;

import android.widget.ImageView;

import pl.orbitemobile.wspolnoty.BasePresenter;
import pl.orbitemobile.wspolnoty.BaseView;

public class ArticleContract {
    
    interface View extends BaseView<Presenter> {
        
        void setDescritpion(String description);
        
        void setTitle(String title);
        
        void setImgUrl(String imgUrl);
        
        void setAppbarCollapsingImage(final ImageView appbarCollapsingImage);
        
        void showErrorMessage();
        
        void showLoadingScreen();
        
        void showArticleDetails();
    }
    
    interface Presenter extends BasePresenter {
        
        void onRetryClick();
    }
}

File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/article/ArticleContract.kt
/*
 * Copyright (c) 2017. All Rights Reserved. Michal Jankowski orbitemobile.pl
 */

package pl.orbitemobile.wspolnoty.activities.article

import android.content.Intent
import io.reactivex.Single
import io.reactivex.SingleObserver
import pl.orbitemobile.mvp.MvpPresenter
import pl.orbitemobile.mvp.MvpView
import pl.orbitemobile.wspolnoty.activities.mvp.DownloadView
import pl.orbitemobile.wspolnoty.data.dto.ArticleDTO

interface ArticleContract {

    abstract class View(layoutId: Int) : MvpView<Presenter>(layoutId), DownloadView { //fixme: download view should extend MVPView
        abstract fun showArticleDetails(article: ArticleDTO)
        abstract fun getIntent(): Intent
    }

    interface Presenter : MvpPresenter<View>, SingleObserver<ArticleDTO> {
        fun onRetryClick()
        fun onShowWebsiteClick()
    }

    interface UseCase {
        fun getRemoteArticleDetails(articleDetailsUrl: String): Single<ArticleDTO>
    }
}